src/components/ModalLedgerResetTokenSignatures.js

```diff
@@ -0,0 +1,176 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useState, useEffect} from 'react';
+import $ from 'jquery';
+import ledger from '../utils/ledger';
+import tokens from '../utils/tokens';
+import { IPC_RENDERER } from '../constants';
+import { t } from 'ttag';
+
+/**
+ * Component shows a modal to reset all token signatures and handle interaction with ledger
+ *
+ * @memberof Components
+ */
+function ModalLedgerResetTokenSignatures() {
+  const modalId = "resetTokenSignatures";
+  const [waitingLedger, setWaitingLedger] = useState(false);
+  const [showOk, setOk] = useState(false);
+  const [errorMessage, setError] = useState(null);
+
+  const closeModal = () => {
+    $(`#${modalId}`).modal('hide');
+  }
+
+  const initialState = () => {
+    setWaitingLedger(false);
+    setError(null);
+    setOk(false);
+  }
+
+  /**
+   * Method called when user clicks the button to reset token signatures
+   *
+   */
+  const handleSend = () => {
+    setError(null);
+    // send command to ledger
+    setWaitingLedger(true);
+    ledger.resetTokenSignatures();
+  }
+
+  /**
+   * Handle the response of a reset token signatures call to Ledger.
+   *
+   * close the modal on success, show error on failure
+   *
+   * @param {IpcRendererEvent} event May be used to reply to the event
+   * @param {Object} arg Data returned from the reset token signatures call
+   */
+  const handleSignatureReset = (event, arg) => {
+    setWaitingLedger(false);
+    if (arg.success) {
+      // clean signatures and close on ok
+      tokens.resetTokenSignatures();
+      // Show ok message
+      setOk(true);
+    } else {
+      switch (arg.error.status) {
+        // user deny
+        case 0x6985:
+          setError('user_deny');
+          break;
+        default:
+          setError('unknown_error');
+      }
+    }
+  }
+
+  /**
+   * Effects
+   */
+
+  // useEffect second parameter [] ensures this runs only once
+  useEffect(() => {
+    $(`#${modalId}`).on('hidden.bs.modal', (e) => {
+      // clean state when closing
+      initialState();
+      // remove ledger listeners when closing
+      if (IPC_RENDERER) {
+        IPC_RENDERER.removeAllListeners("ledger:tokenSignatureReset");
+      }
+    });
+
+    $(`#${modalId}`).on('shown.bs.modal', (e) => {
+      // ledger listeners
+      if (IPC_RENDERER) {
+        IPC_RENDERER.on("ledger:tokenSignatureReset", handleSignatureReset);
+      }
+    });
+
+    return () => {
+      // remove all listeners on unmount
+      $(`#${modalId}`).off();
+    }
+  }, []);
+
+  /**
+   * Render
+   */
+
+  const renderFooter = () => {
+    // Don't render the footer separator if there is no footer
+    if (waitingLedger) return (
+      <div className="modal-footer">
+        <p>{t`Waiting for Ledger...`}</p>
+      </div>
+    );
+
+    return (
+      <div className="modal-footer">
+        {!showOk && <button type="button" className="btn btn-hathor" onClick={handleSend}>{t`Start`}</button>}
```

Comment:
I think it's a little weird that we still show this `Start` button after the case of `User denied on Ledger`.

Shouldn't we change the text to `Try again` if some error happens, instead of `Start`?

Reply:
Done